src/ucs/sys/string.h

```diff
@@ -128,6 +129,22 @@ char* ucs_strncpy_safe(char *dst, const char *src, size_t len);
 char *ucs_strtrim(char *str);
 
 
+/**
+ * Get pointer to file name in path, same as basename but do not
+ * modify source string.
+ *
+ * @param path Path to parse.
+ * 
+  * @return file name
+*/
+static UCS_F_ALWAYS_INLINE const char* ucs_basename(const char *path)
+{
+    const char *name = strrchr(path, '/');
+
+    return (name == NULL) ? path : name + 1;
+}
```

Comment:
This is very naive implementation of basename(3).  I cannot say if it is enough for the purposes of UCX, or do the callers depend on the full POSIX semantic.  But I can point out that, for example,  POSIX requires that NULL or empty string argument returns ".", that "/" and "//" returned "/", and that trailing '/' characters were deleted.

Reply:
this is replacement enough for internal use only, we had a lot of same implementations of this function, I just merged it into single location